src/java/picard/sam/markduplicates/util/OpticalDuplicateFinder.java

```diff
@@ -72,9 +72,12 @@ public OpticalDuplicateFinder(final String readNameRegex, final int opticalDupli
      * unmodified a copy of the list should be passed to this method.
      *
      * @param list a list of reads that are determined to be duplicates of one another
+     * @param keeper a single PhysicalLocation that is the one being kept as non-duplicate, and thus should never be
+     *               annotated as an optical duplicate. May in some cases be null, or a PhysicalLocation not
+     *               contained within the list!
```

Comment:
Got it.  I know you guys have said this below, but I think there needs to be a lot of comments in the code explaining what is happening below.